sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```
@@ -447,13 +439,18 @@ private static String getEntityPath(ClientLogger logger, MessagingEntityType ent
                     new IllegalArgumentException("Unknown entity type: " + entityType));
         }
 
-        return entityPath;
-    }
+        switch (subQueue) {
+            case NONE:
+            case TRANSFER_QUEUE:
+                break;
+            case DEAD_LETTER_QUEUE:
+                entityPath += "/$deadletterqueue";
+                break;
+            default:
+                logger.warning("Unsupported value of subqueue type {}.", subQueue);
```

I would throw instead of just logging it.